src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php

```diff
@@ -94,7 +94,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $targetArg = rtrim($input->getArgument('target'), '/');
 
         if (!$targetArg) {
-            $targetArg = $this->getPublicDirectory($this->getContainer());
+            $targetArg = $this->getPublicDirectory($kernel->getContainer());
```

Comment:
alternatively, we could also just inject the `kernel.project_dir` parameter in the constructor

Reply:
I agree! The `getPublicDirectory()` method only needs the value of `kernel.project_dir`.

Reply:
On the other hand, custom service definitions will then have to be updated to account for the extra conductor argument. We probably rather want to make that move in 4.3 to be able to issue a proper deprecation.

Reply:
let's discuss this in #29840